src/common/components/cards/failed-instances-section.tsx

```diff
@@ -40,7 +41,8 @@ export const FailedInstancesSection = NamedFC<CommonInstancesSectionProps>(
                 visualHelperEnabled={cardsViewData.visualHelperEnabled}
                 allCardsCollapsed={cardsViewData.allCardsCollapsed}
                 outcomeCounter={OutcomeCounter.countByCards}
-                headingLevel={3}
+                headingLevel={titleHeadingLevel + 1}
```

Comment:
imo, it would make more sense for the `ResultSection` to be the component responsible for understanding that the "titleHeadingLevel" should always be one above the content heading level, since it's the component responsible for how those two are composed; maybe a single `headingLevel` prop for the section and have the section know that `titleHeadingLevel = sectionHeadingLevel` and `contentHeadingLevel = sectionHeadingLevel + 1`?

Reply:
Good idea! Moved this logic into `ResultSection`. I also prefer `sectionHeadingLevel` to `titleHeadingLevel`, so I renamed that